llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -1105,6 +1105,81 @@ def IREELinalgExt_WinogradOutputTransformOp : IREELinalgExt_Op<"winograd.output_
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Softmax
+//===----------------------------------------------------------------------===//
+
+class CheckNumOperands<int i> : CPred<"$_op.getNumOperands() == " # i>;
+
+def IREELinalgExt_SoftmaxOp : IREELinalgExt_Op<"softmax",
+    [PredOpTrait<"only one input and one output",  CheckNumOperands<2>>,
+     PredOpTrait<"input and output have same element type", TCopVTEtIsSameAs<0, 1>>,
+     DeclareOpInterfaceMethods<ReifyRankedShapedTypeOpInterface>,
+     DeclareOpInterfaceMethods<TilingInterface,
+      ["getIterationDomain",
+       "getLoopIteratorTypes",
+       "getResultTilePosition",
+       "getTiledImplementation"]>]> {
+  let summary = "Softmax operator";
+  let description = [{
+    This op computes a numerically stable version of softmax for a given tensor.
+    For a given input tensor x and specified dimension d,
+    we first compute the max along that dimension (m). We then compute
+    f(x) = exp(x - m). Then, we sum f(x) along dimension d to get l(x). Finally,
+    we compute the softmax as f(x) / l(x).
+  }];
+
+  let arguments = (ins Variadic<AnyShaped>:$inputs,
+                       Variadic<AnyShaped>:$outputs,
```

Comment:
Why not just do:
` let arguments = (ins AnyShaped:$inputs, AnyShaped:$outputs);`

Reply:
We need to clean up the `LinalgExt` ops in general to avoid this... Requires more surgery... For now this seems fine.